application-collabora-ui/src/main/resources/Collabora/Code/Main.xml

```diff
@@ -203,35 +196,34 @@
     </property>
     <property>
       <code>require(['jquery'], function($) {
+  const collaboraPath ='/rest/collabora/files/';
   $(function() {
-    const server = $("#collaboraServer").data('collaboraServer');
     const fileId = $("#collaboraServer").data('fileId');
-    const userCanWrite = $("#collaboraServer").data('mode') == 'edit';
-    const wopiResourceURL = window.location.origin + '/xwiki/rest/wopi/files/' + fileId
-      + '?userCanWrite=' + userCanWrite;
-    const discoveryURL = XWiki.contextPath + '/rest/discovery';
-    $.getJSON(discoveryURL, $.param({
-      xpage: 'plain',
-      outputSyntax: 'plain',
-      server: server,
-      ext: fileId.split('.').pop(),
+    const userCanWrite = $("#collaboraServer").data('mode') === 'edit';
+    const errorMessage = $("#collaboraServer").data('error');
+    const wopiResourceURL = window.location.origin + '/xwiki' + collaboraPath + encodeURIComponent(fileId);
+    const tokenURL = XWiki.contextPath + collaboraPath + encodeURIComponent(fileId) + '/token';
+    $.getJSON(tokenURL, $.param({
       fileId: fileId
     })).done(function(resp) {
-      const wopiClientUrl = resp['urlSrc'];
-      const accessToken =  resp['token'];
-      const wopiUrl = wopiClientUrl + 'WOPISrc=' + wopiResourceURL;
-      $('#collaboraForm').attr('action', wopiUrl);
+      const fileUrlSrc = resp.urlSrc;
+      const accessToken =  resp.value;
+      const params = $.param({
+        userCanWrite: userCanWrite
+      });
+      const actionURL = fileUrlSrc + 'WOPISrc=' + encodeURIComponent(wopiResourceURL) + '?' + params;
```

Comment:
Isn't ``WOPISrc=`` part of the query string? Then what's with the ``?' + params`` at the end? You can't have two query strings so I'm thinking that maybe you want to add that to the ``wopiResourceURL``, but then it should be inside ``encodeURIComponent`` no?